mod_auth/models.py

```diff
@@ -23,35 +23,98 @@ class User(Base):
     role = Column(Role.db_type())
 
     def __init__(self, name, role=Role.user, email=None, password=''):
+	"""
+	Parametrized constructor for the User model
+
+	:param name: The value of the 'name' field of User model
+	:type name: str
+	:param role: The value of the 'role' field of User model
+	:type role: Role Model
+	:param email: The value of the 'email' field of User model (None by default)
+	:type email: str
+	:param password: The value of the 'password' field of User model ('' by default)
+	:type password: str
+	"""
         self.name = name
         self.email = email
         self.password = password
         self.role = role
 
     def __repr__(self):
+	"""
+	Representation function
+	Represent a User Model by its 'name' Field.
+
+	:return name: Returns the 'name' field of the User model
+	:rtype name: str
+	"""
         return '<User %r>' % self.name
 
     @staticmethod
     def generate_hash(password):
+	"""
+	Generates a Hash value for a password
+
+	:param password: The password to be hashed
+	:type password: str 
+	:return : The hashed password
+	:rtype : str
+	"""
         # Go for increased strength no matter what
         return pwd_context.encrypt(password, category='admin')
 
     @staticmethod
     def create_random_password(length=16):
+	"""
+	Creates a random password of default length 16
+
+	:param length: If parameter is passed, length will be the parameter. 16 by default
+	:type length: int
+	:return : Randomly generated password
+	:rtype : str
+	"""
         chars = string.ascii_letters + string.digits + '!@#$%^&*()'
         import os
         return ''.join(chars[ord(os.urandom(1)) % len(chars)] for i in
                        range(length))
 
     def is_password_valid(self, password):
+	"""
+	Checks the validity of the password
+
+	:param password: The password to be validated
+	:type password: str
+	:return : Validity of password
+	:rtype : boolean
+	"""
         return pwd_context.verify(password, self.password)
 
     def update_password(self, new_password):
+	"""
+	Updates the password to a new one
+
+	:param new_password: The new password to be updated
+	:type new_password: str
+	"""
         self.password = self.generate_hash(new_password)
 
     @property
     def is_admin(self):
+	"""
+	Verifies if a User is the admin
+
+	:return : Compares role of the passed Model with 'Admin' and returns result	
+	:rtype: boolean
+	"""
         return self.role == Role.admin
 
     def has_role(self, name):
+	"""
+	Checks whether the User has a particular role
+
+	:param name: Name of the user
+	:type name: str
+	:return : Checks whether a passed User has a role.
```

Comment:
Not fully ;) We have a user object, and call it like user.has_role('admin'), so this phrase isn't really correct.